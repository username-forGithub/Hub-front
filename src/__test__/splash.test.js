/* eslint-disable linebreak-style */
/* eslint-disable import/extensions */
/* eslint-disable linebreak-style */
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';
import SplashScreen from '../components/Splash/Splash';

describe('Splash screen tests', () => {
  it('renders correctly', () => {
    const tree = render(

      <BrowserRouter>
        <SplashScreen />
      </BrowserRouter>
      ,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Splash page has a sign up link', () => {
    const register = render(
      <>

        <BrowserRouter>
          <SplashScreen />
        </BrowserRouter>

      </>,
    );

    expect(register.findByText('Sign up')).toBeTruthy();
  });

  it('Splash page has a log in link', () => {
    const register = render(
      <>

        <BrowserRouter>
          <SplashScreen />
        </BrowserRouter>

      </>,
    );

    expect(register.findByText('Log in')).toBeTruthy();
  });
});
