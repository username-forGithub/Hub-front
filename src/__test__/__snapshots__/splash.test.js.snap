// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash screen tests renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="splash-screen"
      >
        <div
          class="spash-logo"
        >
          <h1>
            smart tutor app
          </h1>
        </div>
        <div
          class="welcome-text"
        >
          <h2>
            Your modern learning software
          </h2>
        </div>
        <div
          class="register-btns"
        >
          <a
            class="btn login-btn"
            href="/Login"
          >
            LOG IN
          </a>
          <a
            class="btn login-btn"
            href="/register"
          >
            Create New Account
          </a>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="splash-screen"
    >
      <div
        class="spash-logo"
      >
        <h1>
          smart tutor app
        </h1>
      </div>
      <div
        class="welcome-text"
      >
        <h2>
          Your modern learning software
        </h2>
      </div>
      <div
        class="register-btns"
      >
        <a
          class="btn login-btn"
          href="/Login"
        >
          LOG IN
        </a>
        <a
          class="btn login-btn"
          href="/register"
        >
          Create New Account
        </a>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
